src/main/java/core/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package core.dao;
+
+import core.lib.Dao;
+import core.model.Manufacturer;
+import core.storage.Storage;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer add(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+    
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return Optional.ofNullable(Storage.manufacturersStorage.get(id));
+    }
+    
+    @Override
+    public List<Manufacturer> getAll() {
+        return new ArrayList<>(Storage.manufacturersStorage.values());
+    }
+    
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer oldValue = Storage.manufacturersStorage.get(manufacturer.getId());
+        Storage.manufacturersStorage.put(manufacturer.getId(), manufacturer);
+        return oldValue;
```

Comment:
we usually don't work with an old value but it's up to you

Reply:
I thought of this, but IDK what manufacturer to return, as there is no point to return the one we are passing into the method.